### PR TITLE
API cleanup

### PR DESCRIPTION
**Pull request by mezis**
- Makes optional API parameters non-positional to avoid mistakes
- Deprecates redundant `_async` methods in favour of the already-existing `async:`option.
- Deprecates `Routemaster::Receiver` which is redundant with [`Routemaster::Drain::Basic`](https://github.com/deliveroo/routemaster-drain#simply-receive-events-from-routemaster)
- Expands the build matrix

### CODE REVIEW

File: routemaster/client.rb [hunk 1]
```diff
@@ -30,36 +30,49 @@ def configure
...
-        _send_event('create', topic, callback, timestamp, async: true)
+      def created_async(topic, callback, timestamp = nil, t: nil)
+        _warn_timestamp_deprecation(timestamp)
+        _warn_async_deprecation
+        _send_event('create', topic, callback, t: t || timestamp, async: true)
       end
 
-      def updated(topic, callback, timestamp = nil)
-        _send_event('update', topic, callback, timestamp)
+      def updated(topic, callback, timestamp = nil, t: nil)
```
Comment (angetheo) [inline routemaster/client.rb]: the readme says
```
Routemaster::Client.updated('widgets', 'https://app.example.com/widgets/2', async: true)
```

I might be missing something but how do we pass the `async` flag here to the method (and to `_send_event`?) (Same for the other actions)

**Review by angetheo — APPROVED** [review]
One comment about the usage of `async`, else LGTM!

File: routemaster/client.rb (hunk 1 above)
Comment (mezis) [reply routemaster/client.rb]: True. Thanks, will fix!

File: .travis.yml [hunk 2]
```diff
@@ -3,9 +3,14 @@ cache: bundler
 rvm:
   - 2.2.5
   - 2.3.1
+  - 2.4.0
```
Comment (FriedSock) [inline .travis.yml:6]: 👍

File: routemaster/client.rb [hunk 3]
```diff
@@ -30,36 +30,36 @@ def configure
...
-        _send_event('noop', topic, callback, timestamp)
+      def noop(topic, callback, timestamp = nil, t: nil)
+        _send_event('noop', topic, callback, t: t || timestamp)
       end
 
-      def noop_async(topic, callback, timestamp = nil)
-        _send_event('noop', topic, callback, timestamp, async: true)
+      def noop_async(topic, callback, timestamp = nil, t: nil)
+        _send_event('noop', topic, callback, t: t || timestamp, async: true)
       end
```
Comment (FriedSock) [inline routemaster/client.rb]: Would be good to add some tests for this change as it is public interface.

File: routemaster/client.rb [hunk 4]
```diff
@@ -30,36 +30,36 @@ def configure
...
-      def noop(topic, callback, timestamp = nil)
-        _send_event('noop', topic, callback, timestamp)
+      def noop(topic, callback, timestamp = nil, t: nil)
+        _send_event('noop', topic, callback, t: t || timestamp)
       end
 
-      def noop_async(topic, callback, timestamp = nil)
-        _send_event('noop', topic, callback, timestamp, async: true)
+      def noop_async(topic, callback, timestamp = nil, t: nil)
+        _send_event('noop', topic, callback, t: t || timestamp, async: true)
```
Comment (FriedSock) [inline routemaster/client.rb]: Would be good to ad some tests for these changes as they are part of the public interface.

**Pull request merged**